go/vt/vtgate/planbuilder/show.go

```diff
@@ -35,6 +50,71 @@ func buildShowPlan(stmt *sqlparser.Show, vschema ContextVSchema) (engine.Primiti
 	}
 }
 
+func buildShowBasicPlan(show *sqlparser.ShowBasic, vschema ContextVSchema) (engine.Primitive, error) {
+	switch show.Command {
+	case sqlparser.Charset:
+		return showCharset(show)
+	case sqlparser.Collation, sqlparser.Function, sqlparser.Procedure,
+		sqlparser.VariableGlobal, sqlparser.VariableSession:
+		return showSendAnywhere(show, vschema)
+	case sqlparser.Database:
+		ks, err := vschema.AllKeyspace()
+		if err != nil {
+			return nil, err
+		}
+
+		var filter *regexp.Regexp
+
+		if show.Filter != nil {
+			filter = sqlparser.LikeToRegexp(show.Filter.Like)
+		}
+
+		if filter == nil {
+			filter = regexp.MustCompile(".*")
+		}
+
+		rows := make([][]sqltypes.Value, 0, len(ks))
+		for _, v := range ks {
+			if filter.MatchString(v.Name) {
+				rows = append(rows, buildVarCharRow(v.Name))
+			}
+		}
+		return engine.NewRowsPrimitive(rows, buildVarCharFields("Database")), nil
+	case sqlparser.StatusGlobal, sqlparser.StatusSession:
+		return engine.NewRowsPrimitive(make([][]sqltypes.Value, 0, 2), buildVarCharFields("Variable_name", "Value")), nil
+	}
+	return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "BUG: unknown show query type %s", show.Command.ToString())
+
+}
+
+func showSendAnywhere(show *sqlparser.ShowBasic, vschema ContextVSchema) (engine.Primitive, error) {
+	ks, err := vschema.FirstSortedKeyspace()
+	if err != nil {
+		return nil, err
+	}
+	return &engine.Send{
+		Keyspace:          ks,
+		TargetDestination: key.DestinationAnyShard{},
+		Query:             sqlparser.String(show),
+		IsDML:             false,
+		SingleShardOnly:   true,
+	}, nil
+}
+
+func showCharset(show *sqlparser.ShowBasic) (engine.Primitive, error) {
+	fields := buildVarCharFields("Charset", "Description", "Default collation")
+	maxLenField := &querypb.Field{Name: "Maxlen", Type: sqltypes.Int32}
+	fields = append(fields, maxLenField)
+
+	charsets := []string{utf8, utf8mb4}
```

Comment:
Does this mean we only support 2 charsets? Cant we send this query down to any MySQL instance and return its output?

Reply:
This is just about moving code. On VTGate level we are not able to handle it. I think we can send it down.
cc: @deepthi